server/db/analysisReports.js

```diff
@@ -1,21 +1,26 @@
 const AnalysisReport = require('./model/AnalysisReport')
-
-const randomNumbers = (size) => Math.random().toString().substring(2, size + 2)
+const randomNumber = require("random-number-csprng")
 
 const generateUuid = async (db) => {
-  const uuid = `${randomNumbers(4)}-${randomNumbers(4)}-${randomNumbers(2)}`
-  let unique
+  let uuid, unique
+
   do {
+    const high = (await randomNumber(0, 9999)).toString().padStart(4, '0')
+    const mid = (await randomNumber(0, 9999)).toString().padStart(4, '0')
+    const low = (await randomNumber(0, 9999)).toString().padStart(4, '0')
+
+    uuid = `${high}-${mid}-${low}`
```

Comment:
Still quite easy to brute force (I think), what about generating strings like this

```
5nbxKPSZ-LR$-j0q8LpdJyoPX3iylv9TH9O0YZyN9VPpE6_nNTl3k4RkgNTJ7unQ9$1L7DU*BsFPMbgZ-78S
```
I wrote a little utility function that could do it.

```js
const generateKey = function(bits){
  var alphabet = "012345678910abcdefghijklmnopqrstuvwxyz" + 
    "ABCDEFGHIJKLMNOPQRSTUVWXYZ$-_+!*"
  var expectedEntroy = Math.pow(2, bits)
  var key = ''
  while(Math.pow(alphabet.length, key.length) < expectedEntroy){
    key += alphabet[Math.floor(Math.random() * alphabet.length)]
  }
  return key
}

// Will result in at least 512 bits of entropy 
console.log(generateKey(512))
```